src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,322 @@
+import { ModalOverlay } from "../../../../components/client/dialog/ModalOverlay";
+import { Dialog } from "../../../../components/client/dialog/Dialog";
+import { Button } from "../../../../components/server/Button";
+import { InputField } from "../../../../components/client/InputField";
+
+import enterInfoStyles from "./EnterInfo.module.scss";
+import viewStyles from "./View.module.scss";
 
 export type Props = {
   onDataSaved: () => void;
+  onGoBack: () => void;
   user: Session["user"];
 };
 
-export const EnterInfo = (_props: Props) => {
-  const _l10n = useL10n();
+// TODO: Add more sophisticated for location data
+const getIsValidInfo = (value: string) => value !== "";
+
+export const EnterInfo = (props: Props) => {
+  const hasMounted = useRef(false);
+
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [location, setLocation] = useState("");
+  const [dateOfBirth, setDateOfBirth] = useState("");
+
+  const [invalidInputs, setInvalidInputs] = useState<Array<string>>([]);
+
+  const explainerDialogState = useOverlayTriggerState({});
+  const explainerDialogTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    explainerDialogState
+  );
+
+  const confirmDialogState = useOverlayTriggerState({});
+  const confirmDialogTrigger = useOverlayTrigger(
+    { type: "dialog" },
+    confirmDialogState
+  );
+
+  const l10n = useL10n();
+
+  const userDetailsData = [
+    {
+      label: l10n.getString("onboarding-enter-details-label-first-name"),
+      key: "firstName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-first-name"
+      ),
+      value: firstName,
+      displayValue: firstName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setFirstName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-last-name"),
+      key: "lastName",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-last-name"
+      ),
+      value: lastName,
+      displayValue: lastName,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLastName,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-location"),
+      key: "location",
+      type: "text",
+      placeholder: l10n.getString(
+        "onboarding-enter-details-placeholder-location"
+      ),
+      value: location,
+      displayValue: location,
+      errorMessage: l10n.getString(
+        "onboarding-enter-details-input-error-message"
+      ),
+      onChange: setLocation,
+    },
+    {
+      label: l10n.getString("onboarding-enter-details-label-date-of-birth"),
+      key: "dateOfBirth",
+      type: "date",
+      placeholder: "",
+      value: dateOfBirth,
+      displayValue: new Date(dateOfBirth).toLocaleDateString("en-US"),
```

Comment:
TIL that you don't need to instantiate a whole `Intl.DateTimeFormat`, neat!

Might I suggest the following slight deviation from the designs, which keeps it short, but also unambiguous:

```suggestion
      displayValue: new Date(dateOfBirth).toLocaleDateString("en-US", { dateStyle: "medium" }),
```